[docs] Add page for `experimentalStyled()` (#26361)
mui-org/material-ui: docs/src/pages/styles/basics/basics.md

## Using the theme context

> Starting from v5, Material-UI is no longer using JSS as its default styling solution. If you still want to use the utilities exported by `@material-ui/styles`, you will need to provide the `theme` as part of the context. For this, you may use the `ThemeProvider` component available in `@material-ui/styles`, or if you are already using `@material-ui/core`, you should use the one exported in `@material-ui/core/styles`, so that the same `theme` can be available in the core components as well.

Reviewer: ```suggestion
Starting from v5, Material-UI is no longer using JSS as its default styling solution. If you still want to use the utilities exported by `@material-ui/styles`, you will need to provide the `theme` as part of the context. For this, you may use the `ThemeProvider` component available in `@material-ui/styles`, or if you are already using `@material-ui/core`, you should use the one exported from `@material-ui/core/styles`, so that the same `theme` can be available in the core components as well.
```

Also: Same as above regarding "core components"

Author: ```suggestion
Starting from v5, Material-UI is no longer using JSS as its default styling solution. If you still want to use the utilities exported by `@material-ui/styles`, you will need to provide the `theme` as part of the context. For this, you may use the `ThemeProvider` component available in `@material-ui/styles`, or if you are already using `@material-ui/core`, you should use the one exported from `@material-ui/core/styles`, so that the same `theme` can be available for the components coming from '@material-ui/core'.
```

Is it more clear like this?

> Starting from v5, Material-UI is no longer using JSS as its default styling solution. If you still want to use the utilities exported by `@material-ui/styles`, you will need to provide the `theme` as part of the context. For this, you may use the `ThemeProvider` component available in `@material-ui/styles`, or if you are already using `@material-ui/core`, you should use the one exported from `@material-ui/core/styles`, so that the same `theme` can be available for the components coming from '@material-ui/core'.

Reviewer 2: ```suggestion
Starting from v5, Material-UI no longer uses JSS as its default styling solution. If you still want to use the utilities exported by `@material-ui/styles`, you will need to provide the `theme` as part of the context. For this, you can use the `ThemeProvider` component available in `@material-ui/styles`, or, if you are already using `@material-ui/core`, you should use the one exported from `@material-ui/core/styles` so that the same `theme` is available for components from '@material-ui/core'.
```
